### PR TITLE
adds optional reconstruction_ids argument and downloads raw and marke…

### PR DESCRIPTION
**Pull request by MatthewAitken**
…r files as well

Addresses #169

### CODE REVIEW

File: pipeline/user_scripts/latest_reconstructions.py [hunk 1]
```diff
@@ -24,28 +24,59 @@ def scan_all(*args, **kwargs):
 
 
 def main():
-    parser = argparse.ArgumentParser(description="utility for downloading the latest final (transformed) swc for each reconstruction")
-    parser.add_argument("runs_table_name", type=str, help="name of the dynamodb table listing all successful pipeline runs")
-    parser.add_argument("results_path", type=str, help="results will be written into this directory")
+    parser = argparse.ArgumentParser(
+        description="downloads the latest final (transformed) swc "
```
Comment (sgratiy) [inline pipeline/user_scripts/latest_reconstructions.py:28]: Should probably clarify, that the script downloads the latest transformed swc as well as the raw swc and a marker

**Comment by sgratiy** [comment]
I am getting a ResourceNotFound exception
```
(nm) python latest_reconstructions.py nm-stage-deployment-out-runs C:\\morph\\latest_reconstructions --reconstruction_ids 1018012596 101597
1405                                                                                                                                         
Traceback (most recent call last):                                                                                                           
  File "latest_reconstructions.py", line 82, in <module>                                                                                     
    main()                                                                                                                                   
  File "latest_reconstructions.py", line 46, in main                                                                                         
    for item in scan_all(TableName=args.runs_table_name):                                                                                    
  File "latest_reconstructions.py", line 20, in scan_all                                                                                     
    response = dynamo.scan(*args, **kwargs)                                                                                                  
  File "C:\Users\sergeyg\AppData\Local\Continuum\anaconda2\envs\nm\lib\site-packages\botocore\client.py", line 316, in _api_call             
    return self._make_api_call(operation_name, kwargs)                                                                                       
  File "C:\Users\sergeyg\AppData\Local\Continuum\anaconda2\envs\nm\lib\site-packages\botocore\client.py", line 626, in _make_api_call        
    raise error_class(parsed_response, operation_name)                                                                                       
botocore.errorfactory.ResourceNotFoundException: An error occurred (ResourceNotFoundException) when calling the Scan operation: Requested res
ource not found 
                                                                                                              
```

File: pipeline/user_scripts/latest_reconstructions.py [hunk 2]
```diff
@@ -24,28 +24,59 @@ def scan_all(*args, **kwargs):
...
     args = parser.parse_args()
 
     runs = []
     for item in scan_all(TableName=args.runs_table_name):
         runs.append({key: item[key]["S"] for key in item})
     runs = pd.DataFrame(runs)
-        
+
+    if args.reconstruction_ids:
+        runs = runs.loc[runs['ReconstructionId'].isin(
```
Comment (sgratiy) [inline pipeline/user_scripts/latest_reconstructions.py:51]: Should probably give a warning if an id is not in the table.

File: pipeline/user_scripts/latest_reconstructions.py (hunk 1 above)
Comment (sgratiy) [reply pipeline/user_scripts/latest_reconstructions.py:28]: And that the names includes specimen id not the reconstruction id.

**Review by sgratiy — COMMENTED** [review]
Looks good overall, but I get an exception when running

File: pipeline/user_scripts/latest_reconstructions.py [hunk 3]
```diff
@@ -24,28 +24,59 @@ def scan_all(*args, **kwargs):
...
+        description="downloads the latest final (transformed) swc "
+                    "for each reconstruction")
+    parser.add_argument(
+        "runs_table_name",
+        type=str,
+        help="name of the dynamodb table listing all successful pipeline runs")
+    parser.add_argument(
+        "results_path",
+        type=str,
+        help="results will be written into this directory")
```
Comment (sgratiy) [inline pipeline/user_scripts/latest_reconstructions.py:37]: Mention in help that this should be an existing directory

**Review by gnayuy — APPROVED** [review]

**Pull request merged**